inc/migrate-terms.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Migrate groups from WordPress to Hugo data as YAML
+ */
+namespace HH_Hugo;
+
+class Migrate_Terms {
+	/**
+	 * @var array WordPress terms combined from post_tag and category taxonomies
+	 */
+	protected $wp_terms = array();
+
+	/**
+	 * @var array Terms for Hugo tags taxonomy
+	 */
+	protected $hugo_terms = array();
+
+	/**
+	 * @var int Term ID of Uncategorized category
+	 */
+	protected $uncategorized_id;
+
+	/**
+	 * Instantiate
+	 *
+	 * @param int $post_id WordPress post ID
+	 */
+	public function __construct( $post_id ) {
+		if ( defined( 'HH_HUGO_UNIT_TESTS_RUNNING' ) ) {
+			return;
+		}
+		$this->wp_terms = $this->get_wp_terms( $post_id );
+		$this->hugo_terms = $this->get_hugo_terms( $this->wp_terms );
+	}
+
+	/**
+	 * Convert WP terms to Hugo terms
+	 *
+	 * @param array $wp_terms WP term IDs
+	 * @return array Hugo tags, categories, groups
+	 */
+	public function get_hugo_terms( $wp_terms ) {
+		$hugo_terms = array();
+
+		foreach( $wp_terms as $wp_term ) {
+			// get Hugo term data
+			if ( $hugo_term = hh_hugo_get_term_data( $wp_term ) ) {
+
+				// create taxonomy list if needed
```

Comment:
This block isn't necessary. The code that adds the term will create the parent key if it doesn't exist.

If you want to ensure that the taxonomy key exists, even if the value is an empty array, I would recommend moving that logic above the loop and out of the `if` block, and just initializing it to an empty array proactively.